### DISCUSSION TITLE
On MacOS what are default location for "Looking for dynamic libraries"?

### DISCUSSION THREAD

**Discussion by ikus060** [pyinstaller/pyinstaller#8431] [Help]
Hello,

I'm using Pyinstaller to create a MacOS application bundle and I notice the resulting package is missing a dynamic library. Looking closely into the pyinstaller log, I get a warning:
```
WARNING: Cannot find path ./librsync.2.dylib (needed by /Users/vmtest/builds/_tyVgVpZ/0/ikus-soft/minarca/minarca-client/.tox/pyinstaller-mac/lib/python3.10/site-packages/rdiff_backup/_librsync.cpython-310-darwin.so)
```
I'm kinda surprise because the file `librsync.2.dylib` is actually located in `/usr/local/lib` which I believe should be lookup by PyInstaller by default. 

So here my question. What I need to do in my spec file to make sure to include this `librsync.2.dylib` file located in `/usr/local/lib/` on MacOS ?

**Comment by rokm** [comment]
What version of PyInstaller are you using?

And how is the dylib actually referenced by `rdiff_backup/_librsync.cpython-310-darwin.so`? Is it via `rpath`? (What is the output if you run `otool -L` against it?)

**Comment by ikus060** [comment]
Hello @rokm 

Thanks for helping. Yes it's included by `@rpath`.

```
% otool -L ./dist/minarca/rdiff_backup/_librsync.cpython-310-darwin.so
./dist/minarca/rdiff_backup/_librsync.cpython-310-darwin.so:
        @rpath/librsync.2.dylib (compatibility version 2.0.0, current version 2.3.4)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)
```

I'm using pyinstaller 5.13.2. I cannot used pyinstaller>=6 because of other limitation.

**Reply by rokm** [reply]
Then it is likely that `rpath` is not set by the `_librsync.cpython-310-darwin.so` extension (for example, because it is expected to be inherited from the python process' executable, which PyInstaller's analysis does not account for; or because it is loaded via `ctypes` when package is imported, and thus its actual location does not matter), and thus the library path erroneously resolves to `./librsync.2.dylib`. I think this should be fixed in v6. If you need to use v5, you'll have to collect that library yourself  (`--add-binary` or its `.spec` equivalent).

**Comment by ikus060** [comment]
On MacOS, `@rpath` are affected by DYLD_LIBRARY_PATH. If I define DYLD_LIBRARY_PATH should that work with PyInstaller ?

**Reply by rokm** [reply]
No.

**Comment by ikus060** [comment]
Ha ok, that explain it.

I will try to add the file manually using spec file
